oml/retrieval/retrieval_results.py

```diff
@@ -64,14 +78,16 @@ def compute_from_embeddings(
         Args:
             embeddings: The result of inference with the shape of ``[dataset_len, emb_dim]``.
             dataset: Dataset having query/gallery split.
-            n_items_to_retrieve: Number of the closest gallery items to retrieve.
-                                 It may be clipped by gallery size if needed.
+            n_items_to_retrieve: Number of the closest gallery items to retrieve. It may be clipped by
+                gallery size if needed. Note, some queries may get less than this number of retrieved items if they
+                don't have enough gallery items available.
 
         """
         assert len(embeddings) == len(dataset), "Embeddings and dataset must have the same size."
 
         if SEQUENCE_COLUMN in dataset.extra_data:
-            sequence_ids = LongTensor(pd.factorize(dataset.extra_data[SEQUENCE_COLUMN])[0])
+            sequence = pd.Series(dataset.extra_data[SEQUENCE_COLUMN])
+            sequence_ids = LongTensor(pd.factorize(sequence)[0])
```

Comment:
Maybe it's not related to this PR, but I recommend to add param `sort=True` to `pd.factorize`. I worked in sim-api with categories or some handcrafted column and my `df` was recreated in runtime each time depends on params. And each time raw values encoded to different values (I used them later for debugging). Unique values of such "categories" were the same, just with different amounts and orders.

It's not happening if `df` remains the same, but why not ...

```python
import pandas as pd
from random import shuffle

value = [1, 1, 2, 3, 1, 2, 3, 4]
shuffle(value)

df = pd.DataFrame({"raw": value})
df["encode"] = pd.factorize(df["raw"])[0]

df = df.sort_values(by="raw")

print(df)
```

Reply:
Nice! I didn't know about this. And thank you for the snippet. Changed.